src/org/zaproxy/zap/extension/ascanrulesAlpha/HttpOnlySite.java

```diff
@@ -0,0 +1,213 @@
+				getName(), //Name
+				getDescription(), //Description
+				getBaseMsg().getRequestHeader().getURI().toString(), //Original URI
+				null, //Param
+				"", //Attack
+				Constant.messages.getString(MESSAGE_PREFIX+"otherinfo", newUri), //OtherInfo 
+				getSolution(), //Solution
+				"", //Evidence
+				getCweId(), //CWE ID
+				getWascId(), //WASC ID
+				newRequest); //HTTPMessage
+		return;
+	}
+	@Override
+	public void scan() {
+		
+		if (getBaseMsg().getRequestHeader().isSecure()) { //Base request is HTTPS
+			if (log.isDebugEnabled()){
+				log.debug ("The original request was HTTPS, so there is not much point in looking further.");
+			}
+			System.out.println("\n\nBase isnt HTTP\n\n");
+			return;
+		}
+
+		HttpMessage newRequest = getNewMsg();
+		System.out.println(newRequest.getRequestHeader().toString());
+		try{
+			log.info("**"+newRequest.getRequestHeader().getURI());
+			System.out.println(newRequest); //Checking the request
+			String host = newRequest.getRequestHeader().getURI().getHost();
+			String path = newRequest.getRequestHeader().getURI().getPath();
+			newRequest.getRequestHeader().setURI(new URI("https",null, host, 443, path));
+			log.info("**"+newRequest.getRequestHeader().getURI());
+			
+		} catch (URIException e) {
+			log.error("Error creating HTTPS URL from HTTP URL:", e);
+			return;
+		}
+
+		if (isStop()) {
+			if (log.isDebugEnabled()) {
+				log.debug("Scanner "+getName()+" Stopping.");
+			}
+			return;
+		}
+
+		try {
+			String record = "http"; //If any redirection is https record will be changed to https
+			int count = 0;
+			while(count < 10) {
+				sendAndReceive(newRequest, false);
+				System.out.println("\n\nThe request was succesful and it was\n"+newRequest.getRequestHeader().toString());
+				System.out.println("\n\nA response was recieved and it was\n"+newRequest.getResponseHeader().toString());
+				System.out.println("\n\nThe status code is: "+newRequest.getResponseHeader().getStatusCode());
+				String loc = newRequest.getResponseHeader().LOCATION;
+				System.out.println("\n\nThe location is: "+newRequest.getResponseHeader().getHeader(loc));
+				int status = newRequest.getResponseHeader().getStatusCode();
+				String redirect = newRequest.getResponseHeader().getHeader(loc);
+				if(status == HttpStatusCode.MOVED_PERMANENTLY || status == HttpStatusCode.FOUND) {
+					String path = newRequest.getRequestHeader().getURI().getPath();
+					if(redirect.contains("https")) {
+						String newHost = redirect.substring(8, redirect.length());
+					    newRequest.getRequestHeader().setURI(new URI("https", null, newHost, 443, path));
+					    record = "https";
+					    count++;
+					} else if (redirect.contains("http")) {
+						raiseAlert(newRequest);
+						break;
+					}
+				}
+				else {
+					break;
+				}
+			}
+			if(record.equals("http") && count == 10) { //For cases when even the 10th redirection is not https
+				raiseAlert(newRequest);
+			}
+		} catch(NoRouteToHostException e) {
```

Comment:
This catch block can be removed, the exception is caught with the following catch block.